lib/helpers/php-compat.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * A collection of polyfills.
+ * 
+ * This is a subset of what's defined in wp-includes/compat.php
+ * Because we're only interested in PHP8-introduced functions
+ */
+
+if ( ! function_exists( 'str_contains' ) ) {
+	/**
+	 * Polyfill for `str_contains()` function added in PHP 8.0.
+	 *
+	 * Performs a case-sensitive check indicating if needle is
+	 * contained in haystack.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $haystack The string to search in.
+	 * @param string $needle   The substring to search for in the haystack.
+	 * @return bool True if `$needle` is in `$haystack`, otherwise false.
+	 */
+	function str_contains( $haystack, $needle ) {
+		return ( '' === $needle || false !== strpos( $haystack, $needle ) );
+	}
+}
+
+if ( ! function_exists( 'str_starts_with' ) ) {
+	/**
+	 * Polyfill for `str_starts_with()` function added in PHP 8.0.
+	 *
+	 * Performs a case-sensitive check indicating if
+	 * the haystack begins with needle.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $haystack The string to search in.
+	 * @param string $needle   The substring to search for in the `$haystack`.
+	 * @return bool True if `$haystack` starts with `$needle`, otherwise false.
+	 */
+	function str_starts_with( $haystack, $needle ) {
+		if ( '' === $needle ) {
+			return true;
+		}
+		return 0 === strpos( $haystack, $needle );
```

Comment:
What I dislike in this implementation is that `strpos()` will have to scan `strlen($haystack) - strlen($needle)` bytes if the string does not start with `$needle`. The longer the string is, the worse performance will be.

What do you think about

```php
return 0 === strncmp( $haystack, $needle, strlen( $needle ) );
```
?

Reply:
In this case, we can even drop the `if ( '' === $needle )` condition, because `strncmp( $haystack, '', 0 )` will always return 0.

Reply:
Thanks, I adjusted the code.